src/vscode-dts/vscode.proposed.taskStatus.d.ts

```diff
@@ -0,0 +1,42 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+
+declare module 'vscode' {
+
+	export enum TaskEventKind {
+		DependsOnStarted = 'dependsOnStarted',
+		AcquiredInput = 'acquiredInput',
+		Start = 'start',
+		ProcessStarted = 'processStarted',
+		Active = 'active',
+		Inactive = 'inactive',
+		Changed = 'changed',
+		Terminated = 'terminated',
+		ProcessEnded = 'processEnded',
+		End = 'end'
+	}
+
+	export interface TaskStatusEvent {
+		/**
+		 * The task item representing the task for which the event occurred
+		 */
+		readonly execution: TaskExecution;
+
+		/**
+		 * The task event kind
+		 */
+		readonly taskEventKind: TaskEventKind;
+	}
+
+	export namespace tasks {
+
+		/**
+		 * An event that is emitted when the status of a task changes.
+		 */
+		export const onDidChangeTaskStatus: Event<TaskStatusEvent>;
```

Comment:
If we have an event we'll probably want a property too (`TaskExecution.taskStatus: TaskStatus`?)

Reply:
Should we wait to see if that's requested?

Reply:
@connor4312 , is the event sufficient for your purposes ?

Reply:
I discussed this with Karthik and think this would just add complexity. You can have a start and stop `TaskExecution` for example and we'd need to keep track of all of those.

Reply:
I think I received exactly this feedback in past API syncs is the reason I say that